### DISCUSSION TITLE
I would like to exchange data between Bonsai and Unity.

### DISCUSSION THREAD

**Discussion by Inagaki-0228site** [bonsai-rx/bonsai#1609] [Q&A] [answered]
I would like to exchange data between Bonsai and Unity. Photograph the mouse from above using DeepLabCut + Bonsai. Then, output the 2D coordinates of the head in real time and send the coordinate data to Unity. Unity moves the object by rewriting the sent coordinates to the object's coordinates. However, I cannot move the data from Bonsai to Unity. please tell me.

**Comment by RoboDoig** [comment, marked as answer]
There are a couple of options for this:

1. Run Bonsai embedded in Unity, we have developed a .unitypackage for this which can be found at https://github.com/emotional-cities/unity
2. Use a networking library to pass data between the two programs, e.g. 0MQ which is supported in Bonsai: https://www.nuget.org/packages/Bonsai.ZeroMQ/
